File: URI Online Judge/Beginners/1557.java
//Problem:
//https://www.urionlinejudge.com.br/judge/es/problems/view/1557
import java.io.IOException;
import java.util.Scanner;
public class Main {

    public static void main(String[] args) throws IOException {
    int[][] v = new int[15][15];
        Scanner sc = new Scanner(System.in);
        int c=0;
        while (true) {
            int n = sc.nextInt();
            if (n == 0)break;
            if(c!=0) System.out.println("");
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++)
                    v[i][j] = (int) Math.pow(2, i + j);
            }
            int u = O(v[n - 1][n - 1]);
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    int s = O(v[i][j]);
                    if (j == n - 1 && n!=1) {
                        for (int k = 0; k < u-s+ 1; k++)
                                System.out.print(" ");
                        System.out.println(v[i][j]);
                    } else if (j > 0 && j < n - 1) {
                            for (int k = 0; k <= u-O(v[n - 1][j])+O(v[n - 1][j])-s; k++)
                                System.out.print(" ");
                        System.out.print(v[i][j]);
                    } else {
                            for (int k = 0; k < u - s; k++)
                                System.out.print(" ");
                        System.out.print(v[i][j]);
                        if(n==1)System.out.println("");
                    }
                }
            }
            System.out.println("");
        }
    }
    private static int O(int a) {
        return String.valueOf(a).length();
    }
}

File: URI Online Judge/Beginners/1557.cpp
//Problem:
//https://www.urionlinejudge.com.br/judge/es/problems/view/1557
#include<iostream>
#include<math.h>
#include<iomanip>
#include<stdint.h>
using namespace std;
int main (){
  int n[100][100],x,u,a,tamano(int l);
  cin>>x;
  while(x!=0){
    for(int i=0;i<=x-1;i++){
        for(int j=0;j<=(x-1);j++){
            n[i][j]=pow(2,j+i);
        }
    }
    u=n[x-1][x-1];
    u=tamano(u);
    for(int i=0;i<=x-1;i++){
        for(int j=0;j<=(x-1);j++){
            if(j==0){
                if(x<=1){
                cout<<n[i][j]<<endl;
                }else{
                a=n[i][j];
                a=tamano(a);
                for(int k=a;k<u;k++)
                cout<<" ";
                cout<<n[i][j];
                a=n[i][j+1];
                a=tamano(a);
                for(int k=a;k<=u;k++)
                cout<<" ";
            }
            }else if(j!=x-1){
            a=n[i][j+1];
            a=tamano(a);
            cout<<n[i][j];
                for(int k=a;k<=u;k++){
                cout<<" ";
                }
            }else{
            cout<<n[i][j]<<endl;
            }
    }
}
cout<<endl;
  cin>>x;
  }
return 0;
}
int tamano(int a){
 int c=1;
 while ( a>9) {
c++;
a= a/10;
}
return c;
}
